src/vivarium_conic_sqlns/components/observers.py

```diff
@@ -9,3 +13,91 @@ def setup(self, builder):
         super().setup(builder)
         self.causes.append('iron_deficiency')
         self.disability_weight_pipelines['iron_deficiency'] = builder.value.get_value('iron_deficiency.disability_weight')
+
+
+class SampleHistoryObserver:
+
+    configuration_defaults = {
+        'metrics': {
+            'sample_history': {
+                'sample_size': 100,
+                'path': '/dev/null'
+            }
+        }
+    }
+
+    @property
+    def name(self):
+        return "sample_history_observer"
+
+    def __init__(self):
+        self.history_snapshots = []
+        self.sample_index = None
+
+    def setup(self, builder):
+        config = builder.configuration['metrics']['sample_history']
+        self.sample_size = config['sample_size']
+        self.path = pathlib.Path(config['path']).resolve()
+        if self.path.suffix != '.hdf':
+            raise ValueError("metrics: sample_history: path must specify a path to an HDF file.")
+
+        self.clock = builder.time.clock()
+        self.randomness = builder.randomness.get_stream('sample_index')
+
+        self.builder = builder
+        self.population_view = builder.population.get_view(['alive', 'age', 'sex', 'exit_time'])
+
+        self.pipelines = {
+            'iron_deficiency_exposure': builder.value.get_value('iron_deficiency.exposure'),
+            'child_stunting_exposure': builder.value.get_value('child_stunting.exposure'),
+            'child_wasting_exposure': builder.value.get_value('child_wasting.exposure'),
+            'lower_resipratory_infections_incidence_rate': builder.value.get_value('lower_respiratory_infections.incidence_rate'),
+            'diarrheal_diseases_incidence_rate' : builder.value.get_value('diarrheal_diseases.incidence_rate'),
+            'measles_incidence_rate': builder.value.get_value('measles.incidence_rate'),
+            'disability_weight': builder.value.get_value('disability_weight')
+        }
+
+        builder.population.initializes_simulants(self.get_sample_index)
+        builder.event.register_listener('collect_metrics', self.record)
+        builder.event.register_listener('simulation_end', self.dump_history)
+
+    def get_sample_index(self, pop_data):
+        if self.sample_index is None:
```

Comment:
Make a sample proportion, use filter for probability and append to your sampled index.